### PR TITLE
New Resource - sentinel_alert_rule_machine_learning_behavior_analytics

### PR DESCRIPTION
**Pull request by kaovd**
PR implements Azure Sentinel ML Behavior Analytics rules as per #11551
This quite literally just copies from Fusion rules as it has the exact same properties. 

# Reference

- Properties/Spec Ref: https://github.com/Azure/bicep-types-az/blob/9d7f9f5caab4cf0b9645aee3dd6328406f0904ab/generated/securityinsights/microsoft.securityinsights/2019-01-01-preview/types.md#mlbehavioranalyticsalertruleproperties
- API: https://github.com/AzureSDKAutomation/azure-resource-manager-schemas/pull/1487/commits/e31fdb6c651295486de179225dae79bd8a02aa7c

# Test Results
```
2021/05/01 22:51:52 [DEBUG] not using binary driver name, it's no longer needed
2021/05/01 22:51:52 [DEBUG] not using binary driver name, it's no longer needed
=== RUN   TestAccSentinelAlertRuleMLBehaviorAnalytics_basic
=== PAUSE TestAccSentinelAlertRuleMLBehaviorAnalytics_basic
=== RUN   TestAccSentinelAlertRuleMLBehaviorAnalytics_complete
=== PAUSE TestAccSentinelAlertRuleMLBehaviorAnalytics_complete
=== RUN   TestAccSentinelAlertRuleMLBehaviorAnalytics_update
=== PAUSE TestAccSentinelAlertRuleMLBehaviorAnalytics_update
=== RUN   TestAccSentinelAlertRuleMLBehaviorAnalytics_requiresImport
=== PAUSE TestAccSentinelAlertRuleMLBehaviorAnalytics_requiresImport
=== CONT  TestAccSentinelAlertRuleMLBehaviorAnalytics_basic
=== CONT  TestAccSentinelAlertRuleMLBehaviorAnalytics_update
=== CONT  TestAccSentinelAlertRuleMLBehaviorAnalytics_complete
=== CONT  TestAccSentinelAlertRuleMLBehaviorAnalytics_requiresImport
--- PASS: TestAccSentinelAlertRuleMLBehaviorAnalytics_basic (253.75s)
--- PASS: TestAccSentinelAlertRuleMLBehaviorAnalytics_complete (255.03s)
--- PASS: TestAccSentinelAlertRuleMLBehaviorAnalytics_requiresImport (363.84s)
--- PASS: TestAccSentinelAlertRuleMLBehaviorAnalytics_update (563.17s)
PASS
ok      github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/services/sentinel    563.286s
```
image

### CODE REVIEW

File: website/docs/r/azurerm_sentinel_alert_rule_ml_behavior_analytics.html.markdown [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+  display_name               = "(Preview) Anomalous SSH Login Detection"
+  log_analytics_workspace_id = azurerm_log_analytics_workspace.example.id
+}
+
+resource "azurerm_sentinel_alert_rule_ml_behavior_analytics" "examplefromtemplate" {
+  name                       = "example-ml-alert-rule-from-template"
+  log_analytics_workspace_id = azurerm_log_analytics_workspace.example.id
+  alert_rule_template_guid   = data.azurerm_sentinel_alert_rule_template.example.name
+  enabled                    = false
+}
```
Comment (magodo) [inline website/docs/r/azurerm_sentinel_alert_rule_ml_behavior_analytics.html.markdown:44]: Can we remove these to make the example a minimal section?
```suggestion
```

File: website/docs/r/azurerm_sentinel_alert_rule_ml_behavior_analytics.html.markdown [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+
+```
+
+## Arguments Reference
```
Comment (magodo) [inline website/docs/r/azurerm_sentinel_alert_rule_ml_behavior_analytics.html.markdown:48]: Can we replace the `Fusion` below to `Machine Learning Behavior Analytics`?

File: azurerm/internal/services/sentinel/registration.go [hunk 3]
```diff
@@ -40,5 +40,6 @@ func (r Registration) SupportedResources() map[string]*schema.Resource {
 		"azurerm_sentinel_data_connector_microsoft_defender_advanced_threat_protection": resourceSentinelDataConnectorMicrosoftDefenderAdvancedThreatProtection(),
 		"azurerm_sentinel_data_connector_office_365":                                    resourceSentinelDataConnectorOffice365(),
 		"azurerm_sentinel_data_connector_threat_intelligence":                           resourceSentinelDataConnectorThreatIntelligence(),
+		"azurerm_sentinel_alert_rule_ml_behavior_analytics":                             resourceSentinelAlertRuleMLBehaviorAnalytics(),
```
Comment (magodo) [inline azurerm/internal/services/sentinel/registration.go:43]: Can we move this line upwards in alphamatic order?

**Review by magodo — CHANGES_REQUESTED** [review]
@kaovd Thanks for this PR!
I have left some additional mostly minor comments that once addressed this should be good to merge 👍

**Comment by magodo** [comment]
@kaovd Shall we change the resource name a bit to expanding the `ml` to `machine_learning`? (sorry to bring this so late)

**Comment by kaovd** [comment]
> @kaovd Shall we change the resource name a bit to expanding the `ml` to `machine_learning`? (sorry to bring this so late)

No problem

**Comment by magodo** [comment]
@kavod Excellent! LGTM 👍

image

**Comment by favoretti** [comment]
@magodo Hmm, file name is not following the naming convention as it has `azurerm_` in the beginning. I'll fix that with my SDK upgrade PR if that's ok with you?

**Pull request merged**